### PR TITLE
Scrum 4341 - RGD Load and HTP tags synonyms fix

### PR DESCRIPTION
**Pull request by VarunReddy1111**

### CODE REVIEW

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java [hunk 1]
```diff
@@ -118,13 +122,53 @@ public HTPExpressionDatasetAnnotation validateHTPExpressionDatasetAnnotationFmsD
...
+										"queryString": "%s",
+										"useKeywordFields": true,
+										"queryType": "matchQuery"
+									}
+								}
+							}
+						}
+					""".formatted("Data Set Category Tags", categoryTag, categoryTag);
+					Map<String, Object> params = new HashMap<>();
+					try{
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java:154]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'try' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java [hunk 2]
```diff
@@ -118,13 +122,53 @@
...
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java:154]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'{' is not preceded with whitespace.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java [hunk 3]
```diff
@@ -118,13 +122,53 @@
...
+						params = mapper.readValue(query, new TypeReference<Map<String, Object>>() {});
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java:155]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'{' is not followed by whitespace.

Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java:155]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'}' is not preceded with whitespace.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java [hunk 4]
```diff
@@ -118,13 +122,53 @@
...
+					} catch (Exception e) {
+						e.printStackTrace();
+						break;
+					}
+					SearchResponse<VocabularyTerm> searchResponse = vocabularyTermService.searchByParams(new Pagination(), params);
+					if(searchResponse.getTotalResults() == 1) {
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java:161]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java [hunk 5]
```diff
@@ -118,13 +122,53 @@
...
+						VocabularyTerm tag = searchResponse.getSingleResult();
+						if (tag == null) {
+							htpAnnotationResponse.addErrorMessage("categoryTags", ValidationConstants.INVALID_MESSAGE + " (" + categoryTag + ")");
+						} else {
+							categoryTags.add(tag);
+						}
+					} else if (searchResponse.getTotalResults() > 1) {
+						htpAnnotationResponse.addErrorMessage("categoryTags", ValidationConstants.INVALID_MESSAGE + " Multiple Tags found in the Vocabulary " +" (" + categoryTag + ")");
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java:169]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'+' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java [hunk 6]
```diff
@@ -118,13 +122,53 @@ public HTPExpressionDatasetAnnotation validateHTPExpressionDatasetAnnotationFmsD
...
+										"queryString": "%s",
+										"useKeywordFields": true,
+										"queryType": "matchQuery"
+									}
+								}
+							}
+						}
+					""".formatted("Data Set Category Tags", categoryTag, categoryTag);
+					Map<String, Object> params = new HashMap<>();
+					try {
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java:154]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineJavaCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line must not have trailing whitespace

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java [hunk 7]
```diff
@@ -118,13 +122,53 @@
...
+										"useKeywordFields": true,
+										"queryType": "matchQuery"
+									}
+								}
+							}
+						}
+					""".formatted("Data Set Category Tags", categoryTag, categoryTag);
+					Map<String, Object> params = new HashMap<>();
+					try { 
+						params = mapper.readValue(query, new TypeReference<Map<String, Object>>() { } );
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java:155]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>')' is preceded with whitespace.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java [hunk 8]
```diff
@@ -118,13 +122,53 @@ public HTPExpressionDatasetAnnotation validateHTPExpressionDatasetAnnotationFmsD
...
+										"useKeywordFields": true,
+										"queryType": "matchQuery"
+									}
+								}
+							}
+						}
+					""".formatted("Data Set Category Tags", categoryTag, categoryTag);
+					Map<String, Object> params = new HashMap<>();
+					try {
+						params = mapper.readValue(query, new TypeReference<Map<String, Object>>(){ } );
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java:155]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'{' is not preceded with whitespace.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java [hunk 9]
```diff
@@ -118,13 +122,53 @@
...
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java:155]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>')' is preceded with whitespace.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java [hunk 10]
```diff
@@ -117,18 +118,33 @@ public HTPExpressionDatasetAnnotation validateHTPExpressionDatasetAnnotationFmsD
...
-					VocabularyTerm tag = vocabularyTermService.getTermInVocabulary(VocabularyConstants.HTP_DATASET_CATEGORY_TAGS, categoryTag).getEntity();
-					if (tag == null) {
+					Map<String, Object> params = new HashMap<>();
+					params.put("name", categoryTag);
+					params.put("query_operator", "or");
+					params.put("synonyms", categoryTag);
+					SearchResponse<VocabularyTerm> searchResponse = vocabularyTermService.findByParams(new Pagination(), params);
+					boolean added = false;
+					if (searchResponse.getTotalResults() > 0) {
+						for(VocabularyTerm tag : searchResponse.getResults()) {
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java:131]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'for' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java [hunk 11]
```diff
@@ -117,18 +118,33 @@
...
+							if(tag.getVocabulary().getName().equals("Data Set Category Tags") && (tag.getName().equals(categoryTag) || tag.getSynonyms().contains(categoryTag))) {
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java:132]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java [hunk 12]
```diff
@@ -117,18 +118,33 @@
...
+								if(categoryTags.containsKey(categoryTag)) {
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java:133]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java [hunk 13]
```diff
@@ -117,18 +118,33 @@
...
+									htpAnnotationResponse.addErrorMessage("categoryTags", ValidationConstants.INVALID_MESSAGE + " Multiple Tags found in the Vocabulary " + " (" + categoryTag + ")");
+								} else {
+									categoryTags.put(categoryTag, tag);
+									added = true;
+								}
+							}
+						}
+					}
+					if(!added) {
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetAnnotationFmsDTOValidator.java:142]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

**Comment by VarunReddy1111** [comment]
Created a new PR, hence closing this

**Pull request closed without merging**